### PR TITLE
Add support for creating account cash balances from the account detai…

### PR DESCRIPTION
**Pull request by Sonlis**
So the user can update cash balance from the account tab. 
Also set `git-hooks/pre-commit` default mode as executable, otherwise pre-commit hook fail by default.

Draft PR for now, as:
- Still need to match the input fields style with dark mode
- The format used by the date picker is DD/MM/YYYY, while they are displayed above as MM/DD/YYYY (why does this format even exist 👎 ). It seems it is not as trivial to change a date picker format: https://stackoverflow.com/a/71175808
- The balance initial value is hardcoded to 0, but could be nice to instead set the same value as the last cash balance chronologically.

For the 2nd and 3rd points, I will have a look in the coming days, but if you have solutions already feel free to add them. I will need help for the first point however, I am not too sure how to tackle it.

### CODE REVIEW

File: libs/ui/src/lib/account-balances/account-balances.component.html [hunk 1]
```diff
@@ -58,3 +58,48 @@
...
+<form
+  class="gf-form"
+  [formGroup]="accountBalanceForm"
+  (ngSubmit)="onSubmitAccountBalance()"
+>
+  <table class="gf-table w-100">
+    <tr>
+      <td>
+        <mat-form-field class="px-2 mb-3">
+          <input
```
Comment (dtslvr) [inline libs/ui/src/lib/account-balances/account-balances.component.html:71]: Can you use `mat-datepicker`?

See here for an example: https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/portfolio/activities/create-or-update-activity-dialog/create-or-update-activity-dialog.html#L159

File: libs/ui/src/lib/account-balances/account-balances.component.html [hunk 2]
```diff
@@ -58,3 +58,48 @@
...
+            formControlName="date"
+            matInput
+            type="date"
+            value="{{ today | date: 'yyyy-MM-dd' }}"
+          />
+        </mat-form-field>
+      </td>
+      <td>
+        <div class="d-flex justify-content-end">
+          <mat-form-field class="px-2">
```
Comment (dtslvr) [inline libs/ui/src/lib/account-balances/account-balances.component.html:81]: `appearance="outline"`

File: libs/ui/src/lib/account-balances/account-balances.component.html [hunk 3]
```diff
@@ -58,3 +58,48 @@
...
+              type="number"
+            />
+          </mat-form-field>
+        </div>
+      </td>
+      <td class="d-flex justify-content-center">
+        <button
+          class="gf-button"
+          color="primary"
+          mat-raised-button
```
Comment (dtslvr) [inline libs/ui/src/lib/account-balances/account-balances.component.html:95]: ` mat-flat-button`

File: libs/ui/src/lib/account-balances/account-balances.component.ts [hunk 4]
```diff
@@ -53,6 +70,10 @@ export class AccountBalancesComponent implements OnChanges, OnDestroy, OnInit {
     }
   }
 
+  public onSubmitAccountBalance() {
```
Comment (dtslvr) [inline libs/ui/src/lib/account-balances/account-balances.component.ts:73]: Move down to keep the alphabetical order

File: libs/ui/src/lib/account-balances/account-balances.module.ts [hunk 5]
```diff
@@ -14,11 +15,13 @@ import { AccountBalancesComponent } from './account-balances.component';
   exports: [AccountBalancesComponent],
   imports: [
     CommonModule,
+    FormsModule,
     GfValueModule,
     MatButtonModule,
     MatMenuModule,
```
Comment (dtslvr) [inline libs/ui/src/lib/account-balances/account-balances.module.ts:21]: Add `MatFormFieldModule`

**Review by dtslvr — CHANGES_REQUESTED** [review]
I have added a few comments

**Comment by Sonlis** [comment]
Made the changes requested from the above review. Also, the date picker now picks up the local date format, and all inputs match the light or dark mode.

The only thing remaining imo is the appearance of the form, I have not yet been able to make the form match the above table format automatically. I made a small hack for the button to fit on the line: https://github.com/ghostfolio/ghostfolio/pull/3260/files#diff-0147c3e11f8d27c753c051950509bb84464b1f94d66a4e0afa180a219166efcaR97 but that's not ideal, I'll still look for a better way of doing this.

File: apps/api/src/app/account-balance/account-balance.controller.ts [hunk 6]
```diff
@@ -46,4 +48,26 @@ export class AccountBalanceController {
       id
     });
   }
+
+  @HasPermission(permissions.createAccountBalance)
+  @Post()
+  @UseGuards(AuthGuard('jwt'), HasPermissionGuard)
+  public async createAccountBalance(
+    @Body() body: any
```
Comment (dtslvr) [inline apps/api/src/app/account-balance/account-balance.controller.ts:56]: `@Body() data: CreateAccountBalanceDto`

File: apps/api/src/app/account-balance/account-balance.controller.ts [hunk 7]
```diff
@@ -46,4 +48,26 @@ export class AccountBalanceController {
...
+  ): Promise<AccountBalance> {
+    const account = body.Account.connect.id_userId;
+    const data = {
+      Account: {
+        connect: {
+          id_userId: {
+            id: account.id,
+            userId: account.userId
```
Comment (dtslvr) [inline apps/api/src/app/account-balance/account-balance.controller.ts:64]: `userId: this.request.user.id`

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts [hunk 8]
```diff
@@ -140,6 +140,31 @@ export class AccountDetailDialog implements OnDestroy, OnInit {
...
+    date,
+    balance
+  }: {
+    date: string;
+    balance: number;
+  }) {
+    const formattedDate = new Date(date);
+    this.dataService
+      .postAccountBalance({
+        userId: this.user.id,
```
Comment (dtslvr) [inline apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts:153]: Remove from client

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts [hunk 9]
```diff
@@ -140,6 +140,31 @@ export class AccountDetailDialog implements OnDestroy, OnInit {
...
+        accountId: this.data.accountId,
+        balance: balance,
```
Comment (dtslvr) [inline apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts:155]: Abbreviate:
```
balance,
```

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts [hunk 10]
```diff
@@ -140,6 +140,31 @@ export class AccountDetailDialog implements OnDestroy, OnInit {
...
+        currency: this.currency,
```
Comment (dtslvr) [inline apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts:156]: Remove (not needed)

File: libs/ui/src/lib/account-balances/account-balances.component.ts [hunk 11]
```diff
@@ -25,24 +28,38 @@ import { Subject } from 'rxjs';
...
     AccountBalancesResponse['balances'][0]
   > = new MatTableDataSource();
   public displayedColumns: string[] = ['date', 'value', 'actions'];
 
   private unsubscribeSubject = new Subject<void>();
 
-  public constructor() {}
+  public constructor(private dateAdapter: DateAdapter<any>) {}
+
+  public accountBalanceForm = new FormGroup({
```
Comment (dtslvr) [inline libs/ui/src/lib/account-balances/account-balances.component.ts:55]: Move up ⬆️

File: libs/ui/src/lib/account-balances/account-balances.component.ts [hunk 12]
```diff
@@ -12,8 +12,11 @@ import {
   Output,
   ViewChild
 } from '@angular/core';
+import { FormGroup, FormControl, Validators } from '@angular/forms';
+import { DateAdapter, MAT_DATE_LOCALE } from '@angular/material/core';
 import { MatSort } from '@angular/material/sort';
 import { MatTableDataSource } from '@angular/material/table';
+import { format } from 'date-fns';
```
Comment (dtslvr) [inline libs/ui/src/lib/account-balances/account-balances.component.ts:19]: Remove

**Review by dtslvr — CHANGES_REQUESTED** [review]
I have adapted the frontend and added some more comments.

Important: Please refetch the branch due to the force push I did after the rebase!

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts (hunk 8 above)
Comment (Sonlis) [reply apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts:153]: I am not sure what you mean here?

Comment (dtslvr) [reply apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts:153]: Yes, exactly. If you get it from the client, you need to verify it first.

**Review by dtslvr — APPROVED** [review]
Thanks again for all the adaptions, @Sonlis! Ready to merge now.

**Pull request merged**